### PR TITLE
Update release checklist to include minor releases

### PR DESCRIPTION
**Pull request by jonalmeida**
---

We should merge this manually when approved to have the docs regenerated (it didn't seem to work the last time).

### CODE REVIEW

File: docs/contribute/release_checklist.md [hunk 1]
```diff
@@ -20,6 +20,14 @@ permalink: /contributing/release-checklist
 - Wait for taskcluster to complete successfully
 - Check that new versions are on [maven.mozilla.org](https://maven.mozilla.org/?prefix=maven2/org/mozilla/components/):
 
+### Minor releases
+
+Making a minor release has a couple extra steps:
+
+- Create a branch with the format `releases/[major_version].0`, for example, `releases/49.0`.
+- [Cherry-pick](https://git-scm.com/docs/git-cherry-pick) the changes to the new branch.
+- Create the tag and release from this branch instead.
```
Comment (pocmo) [inline docs/contribute/release_checklist.md:29]: Let's also mention bumping the version in `.buildconfig` here. That's what I forgot for dot releases a couple of times... :)

Comment (jonalmeida) [reply docs/contribute/release_checklist.md:30]: Good catch, I forgot to add that to the doc as well. 😄

File: docs/contribute/release_checklist.md [hunk 2]
```diff
@@ -6,20 +6,29 @@ permalink: /contributing/release-checklist
 
 ## Before the release
 
-- Make sure version number is correct or update version in Config.kt 
+- Make sure version number is correct or update version in Config.kt.
```
Comment (Amejia481) [inline docs/contribute/release_checklist.md:9]: Would you mind changing `Config.kt ` for ` buildconfig.yml`? :)

Comment (jonalmeida) [reply docs/contribute/release_checklist.md]: Thanks, done!

**Pull request merged**